Zend/zend_execute.h

```diff
@@ -459,6 +459,45 @@ ZEND_COLD void zend_verify_property_type_error(zend_property_info *info, zval *p
 		} \
 	} while (0)
 
+/* The cache_slot may only be NULL in debug builds, where arginfo verification of
+ * internal functions is enabled. Avoid unnecessary checks in release builds. */
+#if ZEND_DEBUG
+# define HAVE_CACHE_SLOT (cache_slot != NULL)
+#else
+# define HAVE_CACHE_SLOT 1
+#endif
+
+static zend_always_inline zend_class_entry* zend_fetch_ce_from_cache_slot(void **cache_slot, zend_type *type)
```

Comment:
I moved this in the header so that I could reuse it in the JIT, I wanted to keep it in the .c file but I couldn't figure out what I needed to do to still be an inline function, if you have any pointers.

Reply:
It does have to be in a header if you want to make it inline. Having large functions in zend_execute.h isn't great though, as it's included in all files.

Why share this particular bit though? Don't you need basically the whole zend_check_type_slow, modulo some flags? I think it might make more sense to export something like that, and then it's not so important that it be inline, I think.

Reply:
Yeah, did this